Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -790,6 +807,48 @@ void RenderGrid::placeItemsOnGrid(GridTrackSizingAlgorithm& algorithm, std::opti
 #endif
 }
 
+LayoutUnit RenderGrid::masonryContentSize() const
+{
+    return m_masonryLayout.gridContentSize();
+}
+
+static bool isPlacedAtFirstMasonryAxisLine(GridArea area, GridTrackSizingDirection masonryDirection)
```

Comment:
The logic is not masonry specific, so maybe rename to

```cpp
static bool itemGridAreaStartsAtFirstLine(GridArea area, GridTrackSizingDirection direction)
```

Reply:
Changed